src/OrchardCore.Modules/OrchardCore.Liquid/Services/SlugService.cs

```diff
@@ -63,5 +64,10 @@ public string Slugify(string text)
 
             return new string(slug.AsSpan()[..Math.Min(slug.Length, MaxLength)]).Normalize(NormalizationForm.FormC);
         }
+
+        public string Slugify(string text, char hyphen)
+        {
+            throw new NotImplementedException();
```

Comment:
Is this by intention?

Reply:
Yes, I added a method in the interface but we deprecate this class so no need to implement the method here.

Reply:
We deprecate the class but it's still anyone can use it unless you use `true` in the `Obsolete` attribute

Reply:
Yes, just like we do in other places we give time for people to migrate their code.

Reply:
Sure, but my comment is the deprecated class and the new one should be functional, unless you enforce the people to migrate their code immediately by settings `[Obsolete("", true)]`

Reply:
They are both functional, I'm not sure what you mean. Here, the SlugService class was only having one method to return a slug with a hyphen. We don't need to implement the new method with the custom separator because this is an addition that I just made. The real new SlugService is in the OrchardCore project which contains an implementation of the Slugify method with a custom separator. Here, I don't want to implement this method in a class that we will deprecate, which may lead to encouraging people using this class when they shouldn't.